base/dates/io.jl

```diff
@@ -152,7 +152,7 @@ function parse(x::AbstractString,df::DateFormat)
         throw(ArgumentError("Delimiter mismatch. Couldn't find first delimiter, \"$(df.slots[1].transition)\", in date string"))
     end
     periods = Period[]
-    extra = []
+    extra = Period[]
```

Comment:
I also tried to debug this, but I thought that extra should be Any type, as I guess by reading line 159:
```julia
pe != nothing && (isa(pe,Period) ? push!(periods,pe) : push!(extra,pe))
```
This will basically undo the commit https://github.com/JuliaLang/julia/commit/7d94c279f6a5cfa2dadb8085e978968ab138adb2.